### PR TITLE
Tab operations

### PR DESCRIPTION
**Pull request by dariosanp**
minor Changes to AppBuilder including adding instance variables for already existing components along with getters and setters for them, and a new function for splitting tabs. Moved IDEJTabbedPane to view under LeftIDEJTabbedPane whit the addition of RightIDEJTabbedPane and ParentIDEJTabbedPane.

### CODE REVIEW

File: src/main/java/app/IDEAppBuilder.java [hunk 1]
```diff
@@ -155,4 +160,34 @@ public JScrollPane getEditorScrollPane() {
         return editorScrollPane;
     }
 
+    public void splitEditor(JSplitPane newSplit) {
+        leftRightSplitPane.setRightComponent(newSplit);
```
Comment (AriGood) [inline src/main/java/app/IDEAppBuilder.java:164]: Maybe move this to view folder as it is public

File: src/main/java/entity/RightIDEJTabbedPane.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            int currentIndex = getSelectedIndex();
+            if (currentIndex != -1) {
+                JTextArea currentTextArea = editorObjs.get(currentIndex).getTextArea();
+                ideAppBuilder.initializeAutoComplete(AutoCompleteBST.buildWithJavaKeywords(), currentTextArea);
+            }
+        });
+    }
+
+    @Override
+    protected JPopupMenu createPopupMenu(int selectedIndex) {
```
Comment (AriGood) [inline src/main/java/entity/RightIDEJTabbedPane.java:30]: this should likely be moved to the parent class as it is repeated in left

File: src/main/java/entity/LeftIDEJTabbedPane.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            int currentIndex = getSelectedIndex();
+            if (currentIndex != -1) {
+                JTextArea currentTextArea = editorObjs.get(currentIndex).getTextArea();
+                ideAppBuilder.initializeAutoComplete(AutoCompleteBST.buildWithJavaKeywords(), currentTextArea);
+            }
+        });
+    }
+
+    @Override
+    protected JPopupMenu createPopupMenu(int selectedIndex) {
```
Comment (AriGood) [inline src/main/java/entity/LeftIDEJTabbedPane.java:23]: this should likely be moved to the parent class as it is repeated in left

**Review by AriGood — APPROVED** [review]

**Comment by marburcea** [comment]
Reviewed changes and looks good!

**Comment by yaraalkurdi** [comment]
Approved

**Comment by mariana-novoa** [comment]
Approved

**Pull request merged**